scylla-cql/src/errors.rs

```diff
@@ -506,6 +514,21 @@ pub enum BadKeyspaceName {
     IllegalCharacter(String, char),
 }
 
+/// An error that occurred when selecting a node connection
+/// to perform a request on.
+#[derive(Error, Debug, Clone)]
+#[non_exhaustive]
+pub enum ConnectionPoolError {
+    #[error("The pool is broken; Last connection failed with: {last_connection_error}")]
+    Broken {
+        last_connection_error: Arc<dyn Error + Sync + Send>,
+    },
```

Comment:
Could you explain (preferably in commit message too) why this is `Arc<dyn Error + Sync + Send>` and not `ConnectionError`?

Reply:
> From what I recall, there was the problem (which we discussed) about a cyclic dependency of errors that can only be broken by use of `dyn`.

That's true, but the cycle is broken in `BrokenConnectionError` type. I think we could make use of `ConnectionError` here (since it's `Clone` anyway).